File: web/src/api/cmsVisitor.ts
import { http,resData } from "@/utils/http";
 
// @Tags CmsVisitor
// @Summary 创建CmsVisitor
export const createCmsVisitor = (data?: object) => {
   return http.request<resData>("post", "/cmsVisitor/createCmsVisitor", { data });  
}

// @Tags CmsVisitor
// @Summary 删除CmsVisitor
export const deleteCmsVisitor = (data?: object) => {
   return http.request<resData>("post", "/cmsVisitor/deleteCmsVisitor", { data });  
}
 

// @Tags CmsVisitor
// @Summary 删除ByIdsCmsVisitor
export const deleteCmsVisitorByIds = (data?: object) => {
   return http.request<resData>("post", "/cmsVisitor/deleteCmsVisitorByIds", { data });  
}
 

// @Tags CmsVisitor
// @Summary 更新CmsVisitor
export const updateCmsVisitor = (data?: object) => {
   return http.request<resData>("post", "/cmsVisitor/updateCmsVisitor", { data });  
}

// @Tags CmsVisitor
// @Summary 快速编辑
export const quickEdit = (data?: object) => {
   return http.request<resData>("post", "/cmsVisitor/quickEdit", { data });  
}
 

// @Tags CmsVisitor
// @Summary 用id查询CmsVisitor
export const findCmsVisitor = (params?: object) => {
   return http.request<resData>("get", "/cmsVisitor/findCmsVisitor", { params });  
}
 

// @Tags CmsVisitor
// @Summary 分页获取CmsVisitor列表
export const getCmsVisitorList = (params?: object) => {
   return http.request<resData>("get", "/cmsVisitor/getCmsVisitorList", { params });  
}
 

// @Tags CmsVisitor
// @Summary 导出excelCmsVisitor列表
export const excelList = (params?: object) => {
   return http.request<resData>("get", "/cmsVisitor/excelList", { params });  
}

File: server/router/business/cmsVisitor.go
package business

import (
	"go-cms/api/v1"
	"go-cms/middleware"
	"github.com/gin-gonic/gin"
)

type CmsVisitorRouter struct {
}

// InitCmsVisitorRouter 初始化 CmsVisitor 路由信息
func (s *CmsVisitorRouter) InitCmsVisitorRouter(Router *gin.RouterGroup) {
	router := Router.Group("cmsVisitor").Use(middleware.OperationRecord())
	routerNoRecord := Router.Group("cmsVisitor")
	var apiV1 = v1.ApiGroupApp.BusinessApiGroup.CmsVisitorApi
	{
		router.POST("createCmsVisitor", apiV1.CreateCmsVisitor)   // 新建CmsVisitor
		router.POST("deleteCmsVisitor", apiV1.DeleteCmsVisitor) // 删除CmsVisitor
		router.POST("deleteCmsVisitorByIds", apiV1.DeleteCmsVisitorByIds) // 批量删除CmsVisitor
		router.POST("updateCmsVisitor", apiV1.UpdateCmsVisitor)    // 更新CmsVisitor
	    router.POST("quickEdit", apiV1.QuickEdit)  // 快速编辑	
	}
	{
		routerNoRecord.GET("findCmsVisitor", apiV1.FindCmsVisitor)        // 根据ID获取CmsVisitor
		routerNoRecord.GET("getCmsVisitorList", apiV1.GetCmsVisitorList)  // 获取CmsVisitor列表
        routerNoRecord.GET("excelList", apiV1.ExcelList)  // 分页导出excel CmsVisitor列表
	}
}
